src/Compilation.zig

```diff
@@ -962,15 +962,17 @@ pub fn create(gpa: *Allocator, options: InitOptions) !*Compilation {
             break :blk false;
         };
 
-        const darwin_can_use_system_sdk = blk: {
+        const darwin_use_system_sdk = blk: {
             if (comptime !builtin.target.isDarwin()) break :blk false;
-            break :blk builtin.os.tag == .macos and options.target.isDarwin();
+            if (!options.is_native_os) break :blk false;
+            if (builtin.os.tag != .macos or !options.target.isDarwin()) break :blk false;
+            break :blk options.frameworks.len > 0 or options.framework_dirs.len > 0;
         };
 
         const sysroot = blk: {
             if (options.sysroot) |sysroot| {
                 break :blk sysroot;
-            } else if (darwin_can_use_system_sdk) {
+            } else if (darwin_use_system_sdk) {
                 break :blk try std.zig.system.darwin.getSDKPath(arena, options.target);
```

Comment:
I think that this change _might_ fix an issue I've been meaning to file for a while now, https://github.com/ziglang/zig/issues/10217, specifically when specifying a cross-compilation target. WDYT?

Reply:
Yeah, I hope that's the case too actually. Would you mind checking after this gets merged, or building zig directly from this branch?

Reply:
Will do!